src/main/java/com/sonymobile/tools/gerrit/gerritevents/GerritHandler.java

```diff
@@ -84,35 +94,97 @@
      * Creates a GerritHandler with all the default values set.
      *
      * @see GerritDefaultValues#DEFAULT_NR_OF_RECEIVING_WORKER_THREADS
+     * @see GerritDefaultValues#DEFAULT_RECEIVE_THREAD_KEEP_ALIVE_TIME
      */
     public GerritHandler() {
-        this(DEFAULT_NR_OF_RECEIVING_WORKER_THREADS);
+        this(DEFAULT_NR_OF_RECEIVING_WORKER_THREADS, DEFAULT_RECEIVE_THREAD_KEEP_ALIVE_TIME);
     }
 
     /**
-     * Standard Constructor.
+     * Create handler with the given number of maximum worker threads.
      *
      * @param numberOfWorkerThreads the number of event threads.
      */
     public GerritHandler(int numberOfWorkerThreads) {
+        this(numberOfWorkerThreads, DEFAULT_RECEIVE_THREAD_KEEP_ALIVE_TIME);
+    }
+
+    /**
+     * Create handler with the given number of maximum worker threads and given thread keep alive
+     * time (not allowing the thread keep alive to become less
+     * than @see GerritDefaultValues#MIN_RECEIVE_THREAD_KEEP_ALIVE_TIME).
+     *
+     * @param numberOfWorkerThreads the number of event threads.
+     * @param threadKeepAliveTime the number of seconds threads will stay alive.
+     */
+    public GerritHandler(int numberOfWorkerThreads, int threadKeepAliveTime) {
         this.numberOfWorkerThreads = numberOfWorkerThreads;
+        this.threadKeepAliveTime = Math.max(MIN_RECEIVE_THREAD_KEEP_ALIVE_TIME, threadKeepAliveTime);
 
-        workQueue = new LinkedBlockingQueue<Work>();
-        workers = new ArrayList<EventThread>(numberOfWorkerThreads);
-        for (int i = 0; i < numberOfWorkerThreads; i++) {
-            EventThread eventThread = createEventThread("Gerrit Worker EventThread_" + i);
-            eventThread.start();
-            workers.add(eventThread);
-        }
+        startQueue();
     }
 
+
     /**
-     * Create the Event Thread.
-     * @param threadName Name of thread to be created.
-     * @return new EventThread to be used by worker
+     * Returns a factory for creating worker threads for the receiving queue.
+     * Overwrite when you have special requirements, e.g. security constraints
+     *
+     * @return ThreadFactory
      */
-    protected EventThread createEventThread(String threadName) {
```

Comment:
oh, was this removed as well, sorry I missed it. Same binary compatibility problems as the other removed methods I think.